src/Implementations/AzureBlobStorage.cs

```diff
@@ -201,5 +202,37 @@ public async Task<IEnumerable<BlobItem>> GetAllDirectoriesInPathAsync(string pat
 
             return blobs;
         }
+
+        /// <inheritdoc />
+        public async Task DeleteBlobsOlderAync(
+            DateTimeOffset cutOffDate,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            foreach (CloudBlockBlob blockBlob in await GetAllBlobsOlderThanAsync(cutOffDate, cancellationToken))
+            {
+                await blockBlob.DeleteAsync();
+            }
+        }
+
+        private async Task<IEnumerable<CloudBlockBlob>> GetAllBlobsOlderThanAsync(
```

Comment:
This List<T> could get quite long...wouldn't it be more efficient to just delete the blobs as we find that they are too old instead of having to store them in a list and then iterate over that list again?